core/src/main/scala/kafka/server/metadata/ClientQuotaMetadataManager.scala

```diff
@@ -121,16 +121,16 @@ class ClientQuotaMetadataManager(private[metadata] val quotaManagers: QuotaManag
       return
     }
 
-    // Update the cache
-    quotaCache.updateQuotaCache(ipEntity, quotaRecord.key, quotaRecord.value, quotaRecord.remove)
-
     // Convert the value to an appropriate Option for the quota manager
     val newValue = if (quotaRecord.remove()) {
       None
     } else {
       Some(quotaRecord.value).map(_.toInt)
     }
     connectionQuotas.updateIpConnectionRateQuota(inetAddress, newValue)
+
+    // Update the cache
```

Comment:
What's the purpose of moving this code?

Reply:
Good question. I could swear there was a review comment suggesting this, but I can't seem to find it. I believe I moved this to align it with handleUserClientQuota where we update the underlying quota manager and then update the cache. I don't have a strong opinion on which thing happens first, but they should both probably be the same.

Thinking about it more, we might want to be more defensive when calling to the quota managers. If the quota manager cannot be updated, I think we should still update the cache since that reflects the true state of the quota according to the metadata log. Thoughts?

Reply:
Is that code structured in such a way that it can throw?  If so, then yes, I think we want to catch that here.